libr/bin/format/pe/pe.c

```diff
@@ -838,6 +839,69 @@ int PE_(bin_pe_get_actual_checksum)(struct PE_(r_bin_pe_obj_t)* bin) {
 	return computed_cs;
 }
 
+static char* PE_(bin_pe_get_claimed_authentihash)(struct PE_(r_bin_pe_obj_t)* bin) {
+	if (!bin->spcinfo) {
+		return NULL;
+	}
+	RASN1Binary *digest = bin->spcinfo->messageDigest.digest;
+	return r_hex_bin2strdup (digest->binary, digest->length);
+}
+
+static char* PE_(bin_pe_get_actual_authentihash)(struct PE_(r_bin_pe_obj_t)* bin) {
+	const char *hashalgo;
+	const ut8 *block, *hash = NULL;
+	char *hashstr = NULL;
+	ut32 checksum_paddr, security_entry_offset, security_dir_size;
+	ut64 len;
+	PE_(image_data_directory) * data_dir_security;
+	PE_DWord security_dir_offset;
+	RBuffer *buf;
+	RHash *ctx;
+	if (!bin->spcinfo) {
+		return NULL;
+	}
+
+	hashalgo = bin->spcinfo->messageDigest.digestAlgorithm.algorithm->string;
+	checksum_paddr = bin->nt_header_offset + 4 + sizeof(PE_(image_file_header)) + 0x40;
+	security_entry_offset =  bin->nt_header_offset + sizeof(PE_(image_nt_headers)) - 96;
+	data_dir_security = &bin->data_directory[PE_IMAGE_DIRECTORY_ENTRY_SECURITY];
+	security_dir_offset = bin_pe_rva_to_paddr (bin, data_dir_security->VirtualAddress);
+	security_dir_size = data_dir_security->Size;
+
+	buf = r_buf_new ();
```

Comment:
define and assign in the same line, it is more readable (RBuffer *buf = r_buf_new())